pkg/controller/installation/core_controller.go

```diff
@@ -423,7 +423,45 @@ func fillDefaults(instance *operator.Installation) error {
 		}
 	}
 
-	// Default IPAM based on CNI.
+	if instance.Spec.TyphaAffinity == nil {
+		switch instance.Spec.KubernetesProvider {
+		// in AKS, there is a feature called 'virtual-nodes' which represent azure's container service as a node in the kubernetes cluster.
+		// virtual-nodes have many limitations, namely it's unable to run hostNetworked pods. virtual-kubelets are tainted to prevent pods from running on them,
+		// but typha tolerates all taints and will run there.
+		// as such, we add a required anti-affinity for virtual-nodes if running on azure
+		case operator.ProviderAKS:
+			instance.Spec.TyphaAffinity = &operator.TyphaAffinity{
+				NodeAffinity: &operator.NodeAffinity{
+					RequiredDuringSchedulingIgnoredDuringExecution: &v1.NodeSelector{
+						NodeSelectorTerms: []v1.NodeSelectorTerm{{
+							MatchExpressions: []v1.NodeSelectorRequirement{
+								{
+									Key:      "type",
+									Operator: corev1.NodeSelectorOpNotIn,
+									Values:   []string{"virtual-node"},
+								},
+								{
+									Key:      "kubernetes.azure.com/cluster",
+									Operator: v1.NodeSelectorOpExists,
+								},
+							},
+						}},
+					},
+				},
+			}
+		default:
+			instance.Spec.TyphaAffinity = nil
+		}
+	}
+
+	// Default the CalicoNetworkSpec based on the CNI plugin.
+	if instance.Spec.CalicoNetwork == nil {
+		switch instance.Spec.CNI.Type {
+		case operator.PluginCalico:
+			instance.Spec.CalicoNetwork = &operator.CalicoNetworkSpec{}
+		}
+	}
+
```

Comment:
This should not be part of the pick, it wasn't in #1308

Reply:
I remember that's what I did, cherry picked the individual commits frin #1308 , this was part of the PR  https://github.com/tigera/operator/pull/1286 that was in https://github.com/tigera/operator/pull/1308, maybe a bad merge 

removing it here https://github.com/tigera/operator/pull/1368/commits/3ced8136051e91c7bc6762a64d45b3a567dc5423